satellite/admin/user.go

```diff
@@ -174,3 +174,60 @@ func (server *Server) userInfo(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	_, _ = w.Write(data) // nothing to do with the error response, probably the client requesting disappeared
 }
+
+func (server *Server) updateUser(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	vars := mux.Vars(r)
+	userEmail, ok := vars["useremail"]
+	if !ok {
+		http.Error(w, "user-email missing", http.StatusBadRequest)
+		return
+	}
+
+	user, err := server.db.Console().Users().GetByEmail(ctx, userEmail)
+	if errors.Is(err, sql.ErrNoRows) {
+		http.Error(w, fmt.Sprintf("user with email %q not found", userEmail), http.StatusNotFound)
+		return
+	}
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to get user %q: %v", userEmail, err), http.StatusInternalServerError)
+		return
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to read body: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	var input console.User
+
+	err = json.Unmarshal(body, &input)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to unmarshal request: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	if input.FullName != "" {
+		user.FullName = input.FullName
+	}
+	if input.ShortName != "" {
+		user.ShortName = input.ShortName
+	}
+	if input.Email != "" {
+		user.Email = input.Email
+	}
+	if !input.PartnerID.IsZero() {
+		user.PartnerID = input.PartnerID
+	}
+	if len(input.PasswordHash) > 0 {
+		user.PasswordHash = input.PasswordHash
+	}
+
+	err = server.db.Console().Users().Update(ctx, user)
```

Comment:
There's slight potential of overwriting of values, but I guess it would require a different Update method which takes a bunch of pointers as arguments.